autoload/slime.vim

```diff
@@ -292,7 +294,7 @@ function! s:SID()
 endfun
 
 function! s:WritePasteFile(text)
-  " could check exists("*writefile")
+  call mkdir(fnamemodify(g:slime_paste_file, ":p:h"), "p")
```

Comment:
To clarify, my core suggestion is just to add this line, which shouldn't require any changes to the readme or docs. (Forget about all the other stuff changing the default paste file.)

The alternative would be to check `v:shell_error` after the system call to detect whether it errored, and if so emit a warning, ideally including the output of the call in the warning.

Come to think of it, it may be good to do that anyway to catch other failure modes such as insufficient permissions.

Reply:
sounds good: `mkdir` and create, with `v:shell_error` (catching most errors) 👍

Reply:
is doing an `mkdir` on each call expensive?

would checking for the existence of the path enough? 🤔

Reply:
I was wondering about the same thing. My hunch was that, since `mkdir(..., "p")` is supposed to be a no-op when the directory already exists, it probably does its own check internally and returns early, so adding an explicit check would be redundant.

But who cares about hunches? Let's look at the code:

* `mkdir(..., "p")` checks and returns early here: https://github.com/vim/vim/blob/a34b4460c2843c67a35a2d236b01e6cb9bc38734/src/filepath.c#L1505. There is a bit of boilerplate before getting there, mostly normalizing the path and checking which arguments were given, but at least it never enters the recursion or actually tries to create any directories.
* `isdirectory(...)` just performs the check with minimal boilerplate: https://github.com/vim/vim/blob/a34b4460c2843c67a35a2d236b01e6cb9bc38734/src/filepath.c#L1417.

So I guess an explicit check is slightly more economical on the C side of things for the common case where the directory already exists. I have no idea whether the difference would be measurable in light of vimscript function call and branching overheads, and I won't bother profiling, but I'll go ahead and make the change---at least the code will look much less alarming.